### DISCUSSION TITLE
[Feature Request]: Add Self-Hosted SAML or OAuth2/OpenID Provider

### DISCUSSION THREAD

**Discussion by Smallinger** [diced/zipline#321] [Ideas]
Hey would be cool to add Self-Hosted Identity Provider like

i think these are the popular ones with "OAuth2 and OIDC" Support.
[Authentik](https://goauthentik.io/)
[Authelia](https://www.authelia.com/)
[Keycloak](https://www.keycloak.org/)

**Comment by diced** [comment]
This has been suggested before, and I do feel like it would be great to add into Zipline. Unfortunately I'm not current working on any additional OAuth providers for Zipline, although it would be great if could contribute and make a pull request with the feature added.

**Comment by danejur** [comment]
I've created a PR that adds support for Authentik (#372). This pattern could probably be made generic enough to work with other self-hosted OAuth providers, but I only use Authentik so my knowledge of them is limited.

**Comment by TolleyLikesRice** [comment]
+1 on this. I'd like to be able to use my existing Zitadel instance to auth with this. Rather than having fixed setups for each provider, would it not be easier to have just custom oauth provider config options?

**Reply by diced** [reply]
Yes, that's the plan with v4 - currently OIDC via authentik is supported, but it should work with anything that supports OIDC, all that needs to be done is just changing the names to be general

**Comment by diced** [comment]
this shoulve been closed a while ago but oidc has been suported since v4 was released.
